sconfig/shared/src/main/scala/org/ekrich/config/impl/ConfigImplUtil.scala

```diff
@@ -22,6 +22,14 @@ import org.ekrich.config.ConfigSyntax
  * the [[org.ekrich.config]] package.
  */
 object ConfigImplUtil {
+  // investigate why '.' breaks renderRoundTrip, for now I keep it intentionally for backward compatibility
```

Comment:
for now '.' works here only because we enforce
`a.b.c = 1`
to be translated into:
`a: { b: { c: 1} } }`
having optimization to render such single object entries into `a.b.c =1` will be challenging.

Not sure if this '.' walkaround is added to the right abstraction layer.

Reply:
look at `[test02.conf](https://github.com/ekrich/sconfig/pull/466/files#diff-b039769e6e6e84149ddc36eba9a11f7f295277a75936a2c1d053fa134f036970)` I've added some corner cases.

Reply:
Comment is updated as well. Investigation done.